iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp

```diff
@@ -0,0 +1,672 @@
+
+    EXPECT_THAT(clientPortUser.getConnectionState(), Eq(ConnectionState::NOT_CONNECTED));
+}
+
+TEST_F(PortManager_test, DoDiscoveryWithClientConnectResultsInWaitForOfferWhenCallingStopOffer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "45c9cc27-4198-4539-943f-2111ae2d1368");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.connectOnCreate = true;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.offerOnCreate = true;
+
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser = createClient(clientOptions);
+
+    serverPortUser.stopOffer();
+
+    m_portManager->doDiscovery();
+
+    EXPECT_THAT(clientPortUser.getConnectionState(), Eq(ConnectionState::WAIT_FOR_OFFER));
+}
+
+TEST_F(PortManager_test, DoDiscoveryWithClientConnectResultsInWaitForOfferWhenServerIsDestroyed)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "585ad47d-1a03-4599-a4dc-57ea1fb6eac7");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.connectOnCreate = true;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.offerOnCreate = true;
+
+
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser = createClient(clientOptions);
+
+    serverPortUser.destroy();
+
+    m_portManager->doDiscovery();
+
+    EXPECT_THAT(clientPortUser.getConnectionState(), Eq(ConnectionState::WAIT_FOR_OFFER));
+}
+
+TEST_F(PortManager_test, DoDiscoveryWithClientConnectResultsInNoClientsWhenClientIsDestroyed)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3be2f7b5-7e22-4676-a25b-c8a93a4aaa7d");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.connectOnCreate = true;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.offerOnCreate = true;
+
+
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser = createClient(clientOptions);
+
+    EXPECT_TRUE(serverPortUser.hasClients());
+
+    clientPortUser.destroy();
+
+    m_portManager->doDiscovery();
+
+    EXPECT_FALSE(serverPortUser.hasClients());
+}
+
+TEST_F(PortManager_test, CreateMultipleClientsWithConnectOnCreateAndOfferingServerResultsInAllClientsConnected)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "08f9981f-2585-4574-b0fc-c16cf0eef7d4");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.connectOnCreate = true;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.offerOnCreate = true;
+
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser1 = createClient(clientOptions);
+    auto clientPortUser2 = createClient(clientOptions);
+
+    EXPECT_THAT(clientPortUser1.getConnectionState(), Eq(ConnectionState::CONNECTED));
+    EXPECT_THAT(clientPortUser2.getConnectionState(), Eq(ConnectionState::CONNECTED));
+}
+
+TEST_F(PortManager_test,
+       DoDiscoveryWithMultipleClientsNotConnectedAndOfferingServerResultsSomeClientsConnectedWhenSomeClientsCallConnect)
```

Comment:
Ouh ... I think we need snake case for test names ... the title is very expressive (which is good!) but hard to read due to camel case

Reply:
Yeah, we need to write this somewhere down. The question is whether we need to change all the old names